Feature/rake task to add admin

jrussell416 opened the pull request:
This PR addresses this ticket [281 - Rake Task to Add New Admin](https://github.com/wnbrb/wnb-rb-site/issues/281)

The script takes a name and email address, creates the user, and assigns the role of admin to the user.
Password and role are handled as default arguments.

Steps to test:

Run rake 'user:add_user[Name, email]'

Test for this task has been added at spec/tasks/user/add_user_spec.rb

Final commit fixed Rubocop offenses


**Update**
 I squashed  "Move rake task" commit  into my first commit to clean up the history.

I left the frontend updates (271, 291) and the Ubuntu workflow update (298) untouched since they came from merging main


**Update**

Users are now redirected to password reset page to change their password upon log in. The check is on whether the password has ever been changed, rather than on sign in count, since a user can log in, not change the password and log in again and that would result in their default password never being changed.  Once they've changed their password once, future logins will taken them to the admin dashboard .


This also means existing users will be directed to the password change page as well.

Rubocop and erb_lint issues were addressed 

One test was failing after recent changes, so I marked it as pending for future discussion. A second test was updated to account for the two possible paths a user can be redirected to after signing in.

**Update 2**
Testing revealed if a user reset their password before logging in, that password change wouldn't be recognized and they would be forced to change the password again. Newest commit has addressed this so if  user changes their password before logging in for the first time, once they log in they will be taken to the admin dashboard, rather than the password update page

sarah-fflow commented on lib/tasks/new_admin.rake:
```diff
@@ -0,0 +1,26 @@
+      puts
+    end
+  end
+end
```
Minor: no return at end of the file

sarah-fflow commented on spec/models/user_spec.rb:
```diff
@@ -34,4 +34,5 @@
       expect(user).to eq(false)
     end
   end
+
```
Minor: suggest not adding a new line in a file if there are no other changes

sarah-fflow commented on spec/tasks/user/add_user_spec.rb:
```diff
@@ -0,0 +1,23 @@
+  end
+end
+
+end
```
Minor: no new line at end of file
Minor: end tabbing out of alignment

sarah-fflow commented on lib/tasks/new_admin.rake:
```diff
@@ -0,0 +1,26 @@
+      #raw, hashed = Devise.token_generator.generate(User, :reset_password_token)
+      #new_user.reset_password_token = hashed
+      #new_user.reset_password_sent_at = Time.now.utc
+      #Devise::Mailer.reset_password_instructions(new_user, raw)
```
For best practice, we should force a password reset here, if we're using a standard password

And then remove any commented lines that are no longer needed

sarah-fflow commented on spec/tasks/user/add_user_spec.rb:
```diff
@@ -0,0 +1,23 @@
+    expect do
+      Rake::Task['user:add_user'].reenable
+      Rake::Task['user:add_user'].invoke('Admin', 'test@example.com', 'changeme', 'changeme', 'admin')
+      end.to change(User, :count).by(1)
```
Love this neat and succinct test, my specs are much messier!

sarah-fflow reviewed (changes requested):
With apologies because it wasn't clear in the requirement conversations: I believe we should enforce a password reset on first login, and while you're there, I'll be grateful if you would pick up the minor changes too

luciagirasoles commented on app/controllers/admin_controller.rb:
```diff
@@ -4,6 +4,7 @@ class AdminController < ApplicationController
 
   before_action :authenticate_user!
 
+
```
We don't need this extra space here.

luciagirasoles commented on app/controllers/application_controller.rb:
```diff
@@ -1,4 +1,5 @@
 # frozen_string_literal: true
+Rails.logger.info('Loaded ApplicationController')
```
why do we need this here?

sarah-fflow commented on spec/models/user_spec.rb:
```diff
@@ -34,4 +34,4 @@
     end
   end
-end
+end
```
Minor: possibly reinstate the newline at end of file?

jrussell416 commented:
Deleted the non necessary password helper. Created when I generated the new Passwords Controller
Added usage instructions on running the rake task to both the rake file and README. Both changes submitted as part of this PR

sarah-fflow commented on spec/rails_helper.rb:
```diff
@@ -66,3 +66,6 @@
 end
+
+
+Capybara.server = :puma, { Silent: false }
```
Is this here for a reason? Or leftover from debugging? I've approved, because I don't think it does any harm: suggest you remove it either in this PR or in whichever one you pick up next.

sarah-fflow reviewed (approved):
See the comment about debug. It only affects tests, so I've not blocked on it.
